### PR TITLE
chore(datastore): add query & mutate impl

### PR DESCRIPTION
**Pull request by Equartey**
*Description of changes:*
Adds Swift FlutterApiPlugin query and mutate implementations.  

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

### CODE REVIEW

File: packages/amplify_core/lib/src/types/datastore/hub/hub_event_element_with_metadata.dart [hunk 1]
```diff
@@ -35,7 +35,7 @@ class HubEventElementWithMetadata<M extends Model> extends HubEventElement<M> {
   final int version;
 
   /// The last time the model was updated locally, in seconds since epoch.
-  final int lastChangedAt;
+  final int? lastChangedAt;
```
Comment (Jordan-Nelson) [inline packages/amplify_core/lib/src/types/datastore/hub/hub_event_element_with_metadata.dart:38]: This is part of the public API so it is technically a breaking change. Was this intended to be part of the public API? Is this a class that customers use (is there a use case in which they might use it)?

Ideally we should try to find a way to make this non-breaking. If that is not feasible and this was never intended to be public was can discuss deprecating this or marking it as internal.

File: packages/amplify_datastore/example/ios/unit_tests/DataStoreHubEventStreamHandlerTests.swift [hunk 2]
```diff
@@ -63,7 +63,7 @@ class DataStoreHubEventStreamHandlerTests: XCTestCase {
         waitForExpectations(timeout: 1.0)
         // cancellation needed to make sure that Hub token is invalidated to
         // prevent collisions between tests
-        hubHandler.onCancel(withArguments: nil)
+        let _ = hubHandler.onCancel(withArguments: nil)
```
Comment (Jordan-Nelson) [inline packages/amplify_datastore/example/ios/unit_tests/DataStoreHubEventStreamHandlerTests.swift:66]: Q: what is the purpose of this change? Is this the norm in swift or enforced via linting?

File: packages/amplify_datastore/example/ios/Podfile [hunk 3]
```diff
@@ -32,9 +32,9 @@ target 'Runner' do
   use_modular_headers!
 
   # Point pods to internal source files
-  pod 'Amplify', path: '../../ios/internal/'
-  pod 'AWSPluginsCore', path: '../../ios/internal/'
-  pod 'AWSDataStorePlugin', path: '../../ios/internal/'
+  pod 'Amplify', path: '../../ios/internal/', :inhibit_warnings => true
```
Comment (Jordan-Nelson) [inline packages/amplify_datastore/example/ios/Podfile:35]: Q: What is the purpose of this change?

File: packages/amplify_datastore/example/ios/unit_tests/DataStoreHubEventStreamHandlerTests.swift [hunk 4]
```diff
@@ -63,7 +63,7 @@ class DataStoreHubEventStreamHandlerTests: XCTestCase {
         waitForExpectations(timeout: 1.0)
```
Comment (Jordan-Nelson) [inline packages/amplify_datastore/example/ios/unit_tests/DataStoreHubEventStreamHandlerTests.swift:63]: Not related to this PR, but we should remove these from generated files in `.gitattributes`. Everything under `**/example/ios/**` is treated as generated code since the majority of it is generated by flutter during `flutter create`. We should remove `unit_tests` from this.

Could you make that change in this feature branch (can be in a follow up PR) since there will probably be quite a few PRs that will updates native tests?

File: packages/amplify_datastore/lib/src/utils/native_api_helpers.dart [hunk 5]
```diff
@@ -13,6 +14,17 @@ GraphQLRequest<String> nativeRequestToGraphQLRequest(
   );
 }
 
+/// Convert a [GraphQLResponse] to a [NativeGraphQLResponse]
+NativeGraphQLResponse GraphQLResponseToNativeResponse(
```
Comment (Jordan-Nelson) [inline packages/amplify_datastore/lib/src/utils/native_api_helpers.dart:18]:
```suggestion
NativeGraphQLResponse graphQLResponseToNativeResponse(
```

File: packages/amplify_datastore/example/ios/unit_tests/DataStoreHubEventStreamHandlerTests.swift (hunk 2 above)
Comment (Equartey) [reply packages/amplify_datastore/example/ios/unit_tests/DataStoreHubEventStreamHandlerTests.swift:66]: Swift linter complains about the value was not bing used. This is a common pattern to satisfy the linter rule, while denoting the return value is not used. Relevant [SO](https://stackoverflow.com/a/32656172)

File: packages/amplify_datastore/example/ios/Podfile (hunk 3 above)
Comment (Equartey) [reply packages/amplify_datastore/example/ios/Podfile:35]: CI logs were showing tons of lint warnings for these files, but since we they are from Amplify Swift we don't care. We're not going to modify these files. Adding this line prevents warning messages in CI, making those logs more readable.

File: packages/amplify_datastore/example/ios/unit_tests/DataStoreHubEventStreamHandlerTests.swift (hunk 4 above)
Comment (Equartey) [reply packages/amplify_datastore/example/ios/unit_tests/DataStoreHubEventStreamHandlerTests.swift:63]: Good call out, I added that to this branch.

Comment (Jordan-Nelson) [reply packages/amplify_datastore/example/ios/unit_tests/DataStoreHubEventStreamHandlerTests.swift:63]: Thank you

**Review by Jordan-Nelson — APPROVED** [review]
Change LGTM. We may have to make changes to for the breaking change but we can discuss offline and handle that in a follow up.

**Review by khatruong2009 — APPROVED** [review]
